base_local_planner/src/simple_trajectory_generator.cpp

```diff
@@ -206,6 +206,16 @@ bool SimpleTrajectoryGenerator::generateTrajectory(
     return false;
   }
 
+  // make sure we do not exceed max vel theta
+  const double abs_thetav = abs(sample_target_vel[2]);
+  const double turn_radius =  vmag / abs_thetav;
+  const bool spin_in_place = turn_radius < 0.05;
+  if (spin_in_place && abs_thetav > limits_->max_vel_theta_spin
+      || !spin_in_place && abs_thetav > limits_->max_vel_theta)
+  {
+    return false;
+  }
```

Comment:
this is definitely the simplest implementation, but I see two potential problems:
- if `limits->max_vel_theta_spin` is much larger than `limits->max_vel_theta`, then most of the trajectories are rejected, which defeats the purpose of specifying how many samples we want through `vsamples`
- since the `VelocityIterator` just uniformly samples in the [min, max] interval, there are generally no trajectories that have x velocity 0, so depending on the `vsamples` / `acc_lim` params and the `0.05` value here, there is maybe no `spin_in_place` among the candidates.

Hence:
- Either we need to select and test all these parameters and assess the effect
- Or, we change the approach, to for example just manually inserting the trajectories in `SimpleTrajectoryGenerator::initialise`: for example, we could just do sth like
```
if (std::hypot(vel[0], vel[1]) < 0.5) {
    sample_params_.push_back({0, 0, limits_->max_vel_theta_spin);
    sample_params_.push_back({0, 0, -limits_->max_vel_theta_spin);
}
```
at the end of the function, or sth a bit better like
```
if (min_vel[0] < 0 && min_vel[1] < 0 && max_vel[0] > 0 && max_vel[1] > 0) {
    sample_params_.push_back({0, 0, std::min(vel[2] + acc_lim[2] * sim_period_, limits_->max_vel_theta_spin));
    sample_params_.push_back({0, 0, std::max(vel[2] - acc_lim[2] * sim_period_, -limits_->max_vel_theta_spin));
}
```

Reply:
What do you think of the current implementation? I think it strikes a good balance of what we discussed today.
We don't reject trajectories and we don't inject our own samples.
Instead, if there happens to be a spin-in-place trajectory, we simply scale it to use `max_vel_theta_spin` instead of `max_vel_theta`.